modules/parsers/parser-hdbschema/com.sap.xsk.parser.hdbschema/src/main/java/com/sap/xsk/parser/hdbschema/custom/XSKHDBSCHEMACoreListener.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2019-2021 SAP SE or an SAP affiliate company and XSK contributors
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, v2.0
+ * which accompanies this distribution, and is available at
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-FileCopyrightText: 2019-2021 SAP SE or an SAP affiliate company and XSK contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package com.sap.xsk.parser.hdbschema.custom;
+
+import com.sap.xsk.parser.hdbschema.core.HdbschemaBaseListener;
+import com.sap.xsk.parser.hdbschema.core.HdbschemaParser;
+import com.sap.xsk.parser.hdbschema.models.XSKHDBSCHEMADefinitionModel;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import org.antlr.v4.runtime.tree.ParseTree;
+import org.antlr.v4.runtime.tree.ParseTreeProperty;
+
+public class XSKHDBSCHEMACoreListener extends HdbschemaBaseListener {
+
+  private XSKHDBSCHEMADefinitionModel model = new XSKHDBSCHEMADefinitionModel();
+  private ParseTreeProperty<Object> tokens = new ParseTreeProperty<>();
+
+  @Override
+  public void exitHdbschemaDefinition(HdbschemaParser.HdbschemaDefinitionContext ctx) {
+    List<ParseTree> ctxList = ctx.children;
+    for (ParseTree tree : ctxList) {
+      if (tree.getChild(0) instanceof HdbschemaParser.SchemaPropContext) {
```

Comment:
Using getChild(0) is not very predictive.
Instead access the value with direct properties, so the code can look like

```
  ctx.property().forEach(el ->{
        model.setSchema(handleStringLiteral(el.schemaProp().STRING().getText());
      });
```

If you see my previous comment here we should not have a list of schemas, it should be only one shema.